### PR TITLE
perf: Hash git blobs with hardware-accelerated SHA-1

### PR DESCRIPTION
**Pull request by anthonyshew**
## Why

Turborepo's in-process blob hashing (task input hashing for modified/untracked files) used gix's collision-detected SHA-1 (`sha1_checked`, a Rust port of sha1dc). Collision detection costs 5-8x over hardware SHA-1 and buys nothing here: these oids are derived from local working-tree contents to serve as cache-key inputs, and anyone who can plant a crafted collision pair in the repo can already modify the build directly.

The cost is largest when the git index is stat-stale (snapshot-restored workspaces: dev containers, CI workspace caches — every tracked file becomes a hash candidate), and it also prices up the ordinary case of hashing modified files between runs on any machine.

## What

Route `hash_file_maybe_normalized` (the `.git`-present path) through the existing `ManualBlobHasher` (RustCrypto `sha1`, which dispatches to SHA-NI on x86 and the SHA extensions on aarch64 at runtime) instead of `GixBlobHasher`. The gix hasher moves into the test module as a differential oracle.

Outputs are bit-identical for all non-colliding inputs. The CRLF normalization logic is untouched — both hashers already shared it via the `BlobHasher` trait.

## Benchmarks

Interleaved A/B, 8 pairs per scenario, `turbo run build --dry=json` on a 1191-package / 43k-file monorepo (4-core Linux, SHA-NI). "Stale" = stat-stale git index (the snapshot-restore case), "fresh" = normal index.

| scenario | binary | wall (mean) | wall (stdev) | CPU (mean) |
|---|---|---|---|---|
| stale | main | 3703ms | 71ms | 6580ms |
| stale | this PR | 3694ms | 76ms | **5838ms (−11%)** |
| fresh | main | 923ms | 16ms | 2265ms |
| fresh | this PR | 912ms | 33ms | 2246ms |

**Wall time is unchanged** (deltas are within noise): the hashing is parallel and fully overlapped by a serial `git diff HEAD` subprocess that dominates stale-index runs (~2.9s of sha1dc/inflate inside git). This PR is a CPU/contention win and the prerequisite that makes in-process verification of stat-stale entries cheap — a follow-up uses that to skip the serial `git diff` entirely, which is where the wall win lives.

## How to verify

- Existing differential tests assert byte-for-byte agreement with `git hash-object` (raw and `--path`/filters variants, chunk-boundary CRLF cases), covering trailing/standalone CR, chunk-boundary CRLF, and binary auto-detection edge cases.
- Full `--dry=json` output on the 1191-package repo is identical before/after (all task hashes and file hashes; only the random run `.id` differs).

### CODE REVIEW

**Review by tknickman — APPROVED** [review]

File: crates/turborepo-scm/src/repo_index.rs [hunk 1]
```diff
@@ -1337,6 +1516,109 @@ enum EntryClassification {
...
+            //   conversion — safe under any config.
+            // - `is_binary`: git's autocrlf/`text=auto` detection refuses to convert binary
+            //   content, and our NUL check is a conservative subset of git's — safe under
+            //   any config.
+            // - text with CRLFs: safe only if `core.autocrlf` is provably off (otherwise
+            //   git would report the file modified even though the raw bytes match) —
+            //   deferred to a config check at dirty-hash time.
+            let diff_safe = if !exec_matches || outcome.normalized {
+                DiffSafety::Unsafe
+            } else if outcome.crlf_count == 0 || outcome.is_binary {
```
[the remark this answers is not in the document]
Comment (anthonyshew) [reply crates/turborepo-scm/src/repo_index.rs]: Fixed in 5a45b98351: binary content with CRLFs is now `SafeIfEolInert` instead of `Safe` (a forced `text` attribute from an external attributes file eol-converts even binary content), deferring to the `eol_conversion_inert` config check. Removed the now-unused `HashOutcome.is_binary` and added a regression test.
